### ISSUE TITLE
Cannot delete cache files on Windows

### ISSUE THREAD

**Issue by nirvana-msu** [sh4nks/flask-caching#238]
Has this ever been tested on Windows? The following piece of code (repeated in a couple of places) would never work on Windows and raises access error. It is attempting to delete the file, **while** also having it open... Attempt for file deletion should be made outside `with open` block. 
https://github.com/sh4nks/flask-caching/blob/1d33809e7a818cd1b87715bf2dedb3cb44fca780/flask_caching/backends/filesystemcache.py#L174-L178

This issue has been there for a really long time, apparently, but prior to `1.10` release the exception was silently swallowed, whereas now it is being constantly logged.

**Comment by sh4nks** [comment]
I am one of these Linux-only guys and as such I can't test it on Windows.

I pushed a potential fix to: https://github.com/sh4nks/flask-caching/tree/fix-fscache-windows
Could you give it try?

**Comment by filak** [comment]
Thank you for quick fix. I can confirm the patch works. I do not see the "delete key" Win32 errors in IDE anymore.

**Comment by nirvana-msu** [comment]
@sh4nks yes these fixes are good, thanks!

While we're on it, I've had another permission-related issue on Windows - this time a lot more ephimeral. Basicaly it seems that `os.replace` is not guaranteed to work properly on Windows, and could throw errors due to some odd race condition within itself (maybe related to antivirus or something else checking the file).  It's not 100% reproducible - sometimes it works, sometimes it does not.  The line I've had an issue with is this one:
https://github.com/sh4nks/flask-caching/blob/a7ebd599de263a77b4adff5553834ced3be34d51/flask_caching/backends/filesystemcache.py#L221

to workaround my issue I've had to add `os.remove` before it, like
```python
if not is_new_file:
    os.remove(filename)
os.replace(tmp, filename) 
```
There is little I managed to find online on this issue - closest was a similar discussion around how `os.rename` may fail in similar circumstances: https://github.com/conan-io/conan/issues/6560#issuecomment-661679853
